test/unit/code_writer_test.rb

```diff
@@ -563,4 +563,88 @@ def test_write_goto_to_output
 
     assert_equal(expected, output.string)
   end
+
+  def test_write_function_writes_to_output
+    output = StringIO.new
+    code_writer = CodeWriter.new(output)
+    code_writer.write_function("foo", 3)
+    expected = <<~EOF
+      (foo)
+      @0
+      D=A
+      @SP
+      A=M
+      M=D
+      @SP
+      M=M+1
+      @0
+      D=A
+      @SP
+      A=M
+      M=D
+      @SP
+      M=M+1
+      @0
+      D=A
+      @SP
+      A=M
+      M=D
+      @SP
+      M=M+1
+    EOF
+
+    assert_equal(expected, output.string)
+  end
+
+  def test_write_function_writes_to_output
```

Comment:
This is overwriting the previous test because of a small copypasta error:

```suggestion
  def test_write_return_writes_to_output
```

Unfortunately minitest’s style of test definition means this mistake is very easy to make. Ruby is currently [emitting a warning](https://github.com/wildmaples/jack-virtual-machine/pull/17/checks?check_run_id=3818549439#step:7:7) about it, but it’s not causing the tests to fail so it’s pretty well hidden.

In future you might want to consider installing & configuring the [`warning` gem](https://github.com/jeremyevans/ruby-warning) to turn warnings into errors (e.g. in `test_helper.rb`) so that these can’t slip through the net:

```
>> warn 'oh noes'
oh noes
=> nil
>> require 'warning'
=> true
>> Warning.process { :raise }
=> nil
>> warn 'oh noes'
gems/warning-1.2.1/lib/warning.rb:221:in `warn': oh noes (RuntimeError)
```